lrpar/src/lib/parser.rs

```diff
@@ -508,6 +566,29 @@ where
             &lexer.all_lexemes()?[..]
         )?)
     }
+
+    /// Parse input. On success return a parse tree. On failure, return a `LexParseError`: a
+    /// `LexError` means that no parse tree was produced; a `ParseError` may (if its first element
+    /// is `Some(...)`) return a parse tree (with parts filled in by this builder's recoverer).
+    pub fn parse2<K>(
+        &self,
+        lexer: &mut Lexer<StorageT>,
+        actions: Vec<Option<&Fn(Vec<K>) -> K>>,
+        input: String,
```

Comment:
I think we want a `&str` here, so that we don't force users to care about moving the string to us.